### PR TITLE
feat: add sponsor deadline feature

### LINKED ISSUE #471: Hide sponsor section on landing page
Right now the landing page still shows the sponsor section, even though we have enough sponsors this year.
今年は十分なスポンサーがいるにもかかわらず、現時点ではランディング ページにはまだスポンサー セクションが表示されています。

Screenshot 2024-10-26 at 15 30 25

### PR DESCRIPTION
**Pull request by ansh-saini**
Fixes #471

### CODE REVIEW

**Comment by ansh-saini** [comment]
@martinheidegger Can you please review this?

**Comment by ansh-saini** [comment]
Hey @martinheidegger I'll continue to work on this PR, but can you please add the `hacktoberfest-accepted` label on this PR so it can count towards my hacktoberfest contributions?

File: 2024/src/pages/index.tsx [hunk 1]
```diff
@@ -263,7 +266,7 @@ export default function IndexPage() {
       subTitle: t("callForSponsors"),
       available: site.siteMetadata.sponsorFormUrl,
```
Comment (martinheidegger) [inline 2024/src/pages/index.tsx:267]: The `!sponsorDateOverDue` should be in here, shouldn't it?

File: 2024/src/pages/index.tsx [hunk 2]
```diff
@@ -263,7 +266,7 @@ export default function IndexPage() {
...
       render: () =>
-        false ? (
+        !sponsorDateOverDue ? (
```
Comment (martinheidegger) [inline 2024/src/pages/index.tsx:269]: Here I think it should be `site.siteMetadata.sponsorFormUrl` I believe.

**Review by martinheidegger — CHANGES_REQUESTED** [review]
Please see my comments.

File: 2024/src/pages/index.tsx (hunk 1 above)
Comment (ansh-saini) [reply 2024/src/pages/index.tsx:267]: My thought process here was:

If sponsorFormURL is available, show the sponsor section
If sponsorDeadline has passed, show coming soon
If sponsorDeadline is in the future, show the link to the form

If this is not desirable then let me know.

**Comment by ansh-saini** [comment]
@martinheidegger Thank you for the review, I've added some replies to your suggestions.

File: 2024/src/pages/index.tsx (hunk 2 above)
Comment (martinheidegger) [reply 2024/src/pages/index.tsx:269]: The event has three states:

1. There is no sponsor URL: It should show the "coming-soon" state.
2. There is a sponsor URL and it is active: It should show the Sponsor button
3. There is a sponsor URL but the `over-due-date` has passed: nothing should be shown.

This should do the trick:

Screenshot 2024-10-31 at 17 36 31

**Comment by martinheidegger** [comment]
Also, please rebase

**Comment by ansh-saini** [comment]
@martinheidegger Done!

**Review by martinheidegger — APPROVED** [review]
Thanks

**Comment by ansh-saini** [comment]
@martinheidegger Thank you for the quick review and merge. 
Can you please apply the `hacktoberfest-accepted` label to this PR?

**Pull request merged**